src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2574,6 +2578,49 @@ MavlinkReceiver::handle_message_onboard_computer_status(mavlink_message_t *msg)
 	_onboard_computer_status_pub.publish(onboard_computer_status_topic);
 }
 
+void MavlinkReceiver::handle_message_statustext(mavlink_message_t *msg)
+{
+	if (msg->sysid == mavlink_system.sysid) {
+		// log message from the same system
+
+		mavlink_statustext_t statustext;
+		mavlink_msg_statustext_decode(msg, &statustext);
+
+		struct log_message_s log_message;
+
+		switch (statustext.severity) {
+			case MAV_SEVERITY_EMERGENCY:
+			case MAV_SEVERITY_ALERT:
+			case MAV_SEVERITY_CRITICAL:
+				log_message.severity = 0;
+				break;
+
+			case MAV_SEVERITY_ERROR:
+				log_message.severity = 3;
+				break;
+
+			case MAV_SEVERITY_WARNING:
+				log_message.severity = 4;
+				break;
+
+			case MAV_SEVERITY_NOTICE:
+			case MAV_SEVERITY_INFO:
+				log_message.severity = 6;
+				break;
+
+			default:
+				return;
+		}
+
+		log_message.timestamp = hrt_absolute_time();
+
+		strncpy((char *)log_message.text, statustext.text, sizeof(log_message.text));
```

Comment:
```suggestion
		strncpy((char *)log_message.text, statustext.text, sizeof(statustext.text));
```

Reply:
I believe the most robust option is to consider which string is shorter, but to write something like `MIN(sizeof(log_message.text), sizeof(statustext.text))` (because mavlink's string is not guaranteed to be NULL-terminated.

Reply:
Yes, but the first concern is to keep the string operations within the much smaller statustext.text (50 char).